### PR TITLE
Add validation payment params.

### PR DESCRIPTION
**Pull request by ugnku**

### CODE REVIEW

File: src/Gate.php [hunk 1]
```diff
@@ -25,16 +28,25 @@ class Gate
...
+
     /**
      * Gate constructor.
      *
      * @param string $secret Secret key
      * @param string $baseUrl Base URL for concatenate with payment params
+     * @param bool $validateParams Validate payment params before generate PaymentPage URL
      */
-    public function __construct($secret, string $baseUrl = '')
+    public function __construct(string $secret, string $baseUrl = '', bool $validateParams = true)
```
Comment (Ol0ff) [inline src/Gate.php]: I don't think passing validateParams marker to the constructor at this point is a good idea.

File: src/Gate.php [hunk 2]
```diff
@@ -73,4 +90,39 @@ public function handleCallback(string $data): Callback
...
+
+        foreach ($headers as $header) {
+            if (preg_match('/^HTTP\/\d.\d (\d+) /', $header, $match)) {
+                $status = (int) $match[1];
+                break;
+            }
+        }
+
+        if ($status !== 200) {
+            $data = json_decode(stream_get_contents($stream));
```
Comment (Ol0ff) [inline src/Gate.php:126]: Can we make throw new ValidationException in this place? After than, remove unnecessary variables.

File: src/Gate.php (hunk 1 above)
Comment (ugnku) [reply src/Gate.php]: Move control flag into public method setValidationParams(boll)

File: src/Gate.php (hunk 2 above)
Comment (ugnku) [reply src/Gate.php:126]: No, we can not do that, because we need to close stream before throw exception.

**Pull request merged**